### DISCUSSION TITLE
Error while testing WIKI Updater task

### DISCUSSION THREAD

**Discussion by khopithan** [rfennell/AzurePipelines#1317] [Q&A] [answered]
Hi @rfennell,

Thank you for this awesome WIKI updater task plugin. I am getting the below error when I run the pipeline. Please help me to sort out this. Thank you.

**Task**
 - task: WikiUpdaterTask@1
                inputs:
                  repo: 'https://dev.azure.com/MyOrg/DemoProj/_wiki/wikis/DemoProject/777/Release-Notes'
                  branch: 'devops_test'
                  filename: 'xPlatReleaseNotes/releasenotes.md'
                  replaceFile: false
                  appendToFile: true
                  dataIsFile: true
                  sourceFile: '$(Build.ArtifactStagingDirectory)/releasenotes.md'
                  message: 'Update from Build'
                  gitname: 'builduser'
                  gitemail: 'build@demo'
                  useAgentToken: true
                  localpath: '$(Build.ArtifactStagingDirectory)/repo'

**Error**
##[error]Error: Cloning into '/Users/runner/work/1/a/repo'...
bad input: XzT2LZ9e-A58MKsk8V7LB09iLNVxeVe9KpMoK5U-tzHEMvgdFL0xTMrccQadRAZ52pwyqDfU3swbFozvdMm7W4bRG92IWiQZyET2kYrwQtLsx8yGWf6LP-8AMw
fatal: unable to update url base from redirection:
  asked for: https://buildagent:***@dev.azure.com/MyOrg/DemoProj/_wiki/wikis/DemoProject/777/Release-Notes/info/refs?service=git-upload-pack
   redirect: https://spsprodweu4.vssps.visualstudio.com/_signin?realm=dev.azure.com&reply_to=https%3A%2F%2Fdev.azure.com%2FMyOrg%2FDemoProj%2F_wiki%2Fwikis%2FDemoProject%2F777%2FRelease-Notes%2Finfo%2Frefs%3Fservice%3Dgit-upload-pack&redirect=1&hid=1bb714ae-9876-438a-8102-6ee43bb355b5&context=eyJodCI6MiwiaGlkIjoiOWQ2MDUxM2UtMzQyMi00MDM2LWEyYzMtOWEyYjZlYmM1OGM4IiwicXMiOnt9LCJyciI6IiIsInZoIjoiIiwiY3YiOiIiLCJjcyI6IiJ90#ctx=eyJTaWduSW5Db29raWVEb21haW5zIjpbImh0dHBzOi8vbG9naW4ubWljcm9zb2Z0b25saW5lLmNvbSIsImh0dHBzOi8vbG9naW4ubWljcm9zb2Z0b25saW5lLmNvbSJdfQ2

Regards,
Khopi

**Comment by rfennell** [comment, marked as answer]
Your Repo URL looks wrong, it appears to have folder paths in it. 

There is a detailed section on how to find the  correct URL and the how to set the security so the build agent service can update the repo are in [this projects WIKI](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks). The same content is in the [Azure Marketplace listing for the extension](https://marketplace.visualstudio.com/items?itemName=richardfennellBM.BM-VSTS-WIKIUpdater-Tasks) 

An extract from the documentation is as follows 

> ### Azure DevOps Services & Azure DevOps Server (TFS) WIKIs
> 
> The URL to clone a Azure DevOps WIKIs is not obvious.
> 
> ```
> IT IS NOT THE URL SHOWN IN THE BROWSER WHEN YOU VIEW THE WIKI e.g:
> 
> https://dev.azure.com/richardfennell/Git%20project/_wiki/wikis/Git-project.wiki/1/Home
> 
> SO DON'T USE THIS FORM
> ```
> To find the correct URL to clone the repo, and use it as the parameter for this task
> 
> 1. Load the WIKI in a browser
> 1. At the top of the menu pane there is a menu (click the ellipsis ...)
> 1. Select the 'Clone repo' option
> 1. You will get a URL in the form https://richardfennell@dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki. This is the URL needed
> 

Also make sure you read the documentation on authentication, but default the build agent service account has no permissions to update the WIKI, you have to grant .

**This is an extension where reading the documentation is vital as there are plenty of gotchas**

A did a DDD Conference session on the whole process of generating release notes and updating a WIKI. You can find the [video here](https://www.youtube.com/watch?v=xaV3dFoQdV8&t=332s)

**Reply by khopithan** [reply]
Thank you so much @rfennell. It's working fine. I appreciate you for spending time 👏👍
